src/formats/notion-api/block-converter.ts

```diff
@@ -0,0 +1,256 @@
+/**
+ * Block conversion functions for Notion API importer
+ * Converts Notion blocks to Markdown format
+ */
+
+import { BlockObjectResponse, Client } from '@notionhq/client';
+import { ImportContext } from '../../main';
+import { fetchAllBlocks } from './api-helpers';
+
+/**
+ * Convert Notion blocks to Markdown
+ */
+export async function convertBlocksToMarkdown(
+	blocks: BlockObjectResponse[], 
+	ctx: ImportContext,
+	currentFolderPath: string,
+	client: Client,
+	indentLevel: number = 0
+): Promise<string> {
+	const lines: string[] = [];
+	
+	for (let i = 0; i < blocks.length; i++) {
+		if (ctx.isCancelled()) break;
+		
+		const block = blocks[i];
+		const markdown = await convertBlockToMarkdown(block, ctx, currentFolderPath, client, indentLevel);
+		if (markdown) {
+			lines.push(markdown);
+		}
+	}
+	
+	// Join blocks with appropriate spacing
+	// List items should be separated by single newline, other blocks by double newline
+	const result: string[] = [];
+	for (let i = 0; i < lines.length; i++) {
+		result.push(lines[i]);
+		
+		if (i < lines.length - 1) {
+			const currentBlock = blocks[i];
+			const nextBlock = blocks[i + 1];
+			
+			// Check if both current and next are list items
+			const currentIsList = currentBlock.type === 'bulleted_list_item' || currentBlock.type === 'numbered_list_item';
+			const nextIsList = nextBlock.type === 'bulleted_list_item' || nextBlock.type === 'numbered_list_item';
+			
+			// Use single newline between list items, double newline otherwise
+			if (currentIsList && nextIsList) {
```

Comment:
Maybe instead, when adding items to lines, if the item being added and the last item in lines are both lists, then add an empty string to lines. Then this section can be fully replaced with a `.join('\n')`